_posts/2023-12-05-improving-document-retrieval-with-sparse-semantic-encoders.md

```diff
@@ -0,0 +1,482 @@
+
+The following images show example results of dense and sparse encoding.
+
+<table style="border:none">
+  <tr>
+    <td style="border:none">
+        <img height="100%" src="/assets/media/blog-images/2023-12-05-improving-document-retrieval-with-spade-semantic-encoders/embedding.png" />
+    </td>
+    <td style="border:none">
+        <img height="100%" src="/assets/media/blog-images/2023-12-05-improving-document-retrieval-with-spade-semantic-encoders/expand.png" />
+    </td>
+  </tr>
+</table>
+
+_**Left**: Dense vector semantic space. **Right**: Sparse vector semantic space._
+
+## Sparse encoders use more efficient data structures
+
+In dense encoding, documents are represented as high-dimensional vectors. To search these documents, you need to use a k-NN index as an underlying data structure. On the other hand, sparse search can use a native Lucene index because sparse encodings are similar to term vectors used by keyword-based matching. 
+
+Compared to k-NN indexes, **sparse embeddings have the following cost-reducing advantages**: 
+
+1. Much smaller index size.
+1. Reduced runtime RAM cost.
+1. Lower computational cost. 
+
+For a detailed comparison, see [Table II](#table-ii-speed-comparison-in-terms-of-latency-and-throughput).
+
+## Sparse encoders perform better on unfamiliar datasets
+
+In our previous [blog post](https://opensearch.org/blog/semantic-science-benchmarks), we mentioned that searching with dense embeddings presents challenges when encoders are facing unfamiliar content. When an encoder trained on one dataset is used on a different dataset, the encoder often produces unpredictable embeddings, resulting in poor search result relevance. 
+
+Often, BM25 performs better than dense encoders on BEIR datasets that incorporate strong domain knowledge. In these cases, sparse encoders fall back on keyword-based matching, ensuring that their search results are no worse than BM25 ones. For a comparison of search result relevance benchmarks, see [Table I](#benchmarking-results).
+
+## Out of sparse encoders, document-only encoders are the most efficient
+
+You can run a neural sparse search in two modes: **bi-encoder** and **document-only**.
+
+In bi-encoder mode, both documents and search queries are passed through deep encoders. In document-only mode, documents are still passed through deep encoders, but search queries are instead tokenized. In this mode, document encoders are trained to learn more synonym association in order to increase recall. By eliminating the online inference phase, you can **save computational resources** and **significantly reduce latency**. For benchmarks, compare the `Neural sparse doc-only` column with other columns in [Table II](#table-ii-speed-comparison-in-terms-of-latency-and-throughput). 
```

Comment:
search queries are just simply tokenized instead. <we should emphasize that tokenization is much hight-weight>